server/models/user.js

```diff
@@ -49,5 +51,41 @@ module.exports = (sequelize) => {
     }
   });
 
+  User.prototype.generateToTPSecret = async function (email, transporter) {
+    const secret = new OTPAuth.Secret();
+    // new TOTP object using the secret key
+    const totp = new OTPAuth.TOTP({
+      secret: secret.base32,
+      issuer: 'Routed',
+      period: -1,
+      digits: 6,
+    });
+    console.log('TOTP Secret: ', secret.base32);
+    // save the secret key to the session
+    // generate secret token
+    const token = totp.generate();
+    // Save token and expiration timestamp in DB (Expires in 15 minutes from inital Log In)
+    // save totp secret and expiration timestamp in DB
+
+    await this.update({ twoFactorData: { totptimestamp: Date.now() + 900000, totptoken: token } });
+    await this.save();
+
+    // send email with token
+    sendMail(
+      transporter,
+      email,
+      'Your Authentication Code from Routed',
+      `This is your Authentication Code: ${token} . It will expire in 15 minutes.`
+    );
+  };
+
+  User.prototype.verifyTwoFactor = function (req) {
+    const token = req.body.code;
+    const totptoken = this.dataValues.twoFactorData.totptoken;
+    const totptimestamp = this.dataValues.twoFactorData.totptimestamp;
+    const verified = token === totptoken && Date.now() < totptimestamp;
+    return verified;
+  };
+
```

Comment:
Instead of `User.prototype.` syntax, let's use JS class syntax. These can be implemented inside the class definition above, just like toJSON().